### PR TITLE
Add documentation for TODOs

### LINKED ISSUE #1557: Feature: Deprecate old pending functionality in favor of new todo feature
This library already has a feature that allows for `pending` violations to be identified, by module, and converts them to warnings when considering whether to error or not. 

This [spec](https://github.com/ember-template-lint/ember-template-lint-pending-utils/blob/master/docs/spec.md) proposes a more robust set of functionality when dealing with pending items, starting with renaming pending to **todo**. 

From the **Motivation** section in that spec:

> Linting is a fundamental tool to help ensure the quality of a codebase. Ensuring there are as few linting errors as
> possible (ideally 0), is a useful measure of a baseline of code hygiene.
> 
> It's common to leverage linting not just for syntax adherence, but also to direct developers to employ standardized patterns. As
> such, it's a fairly routine activity to introduce new lint rules into a codebase. This introduction, while necessary, can cause
> unintended friction, such as
> 
> - new lint errors being introduced where they previously didn't exist
> - causing unintended delays to shipping new fixes and features
> - an "all or nothing" approach, where new rules require fixing before rollout.
> 
> Having the ability to identify violations as `todo` would allow for this incremental roll out, while providing tools that will allow maintainers to view the full list of todo violations. Todo violations can include a createdDate, allowing thresholds of time to trigger due date violations. This can assist in prioritizing lint rule violation fixes, and can ensure the todo list doesn't remain stagnant.

The rest of that spec details the intended behavior. This issue will detail the changes needed in `ember-template-lint` in order to implement this behavior as first-party functionality.

CHANGES:

- [x] Detect existing `pending` behavior, and ensure correct branching to support both old `pending` and new `todo`
- [x] Rename `@ember-template-lint/pending-utils` to `@ember-template-lint/todo-utils`
- [x] Leverage [`@ember-template-lint/pending-utils`](https://github.com/ember-template-lint/ember-template-lint-pending-utils) to parse and generate `.lint-todo` dir
- [x] Update docs for the new feature
- [x] Ensure adequate test coverage of expected workflows
- [x] Set `--print-pending` option to `hidden` in help

WORKFLOWS:

- when upgrading to new ember-template-lint version that includes todo:

	```
	# behaves as normal - no todo functionality
	ember-template-lint
	```

- when linting failure exists (severity 2) that has a current todo item, switch linting failure's severity to not trigger a failure
	- todo items should _not_ be included in default output (#1248)
	- todo item counts _should_ be included in counts output

		```shell
		✖ 8 problems (5 errors, 3 warnings, 2 todo)
		```	

		if no problems, no output (as normal)
		if no problems _but_ `--include-todo`, output todo count

		```shell
		✓ 0 problems (0 errors, 0 warnings, 2 todo)
		```			

	- todo items should be included with output _if_ running with the `--include-todo` option

		```shell
		ember-template-lint . --include-todo
		```

- when opting in to todo functionality:

	```shell
	# add new todo items for new errors
	# remove todo items for errors that are no longer present
	# existing errors that have todo items should not be modified

	ember-template-lint . --update-todo

	# or 

	ember-template-lint lib/my-inrepo-addon --update-todo
	```

- when you have fixed a todo item, to update todo list:

	```shell
	# only removes todo items
	ember-template-lint . --fix
	```

- when you have deleted a linted file (foo.hbs), todo files associated with that file become orphaned. If you run linting for a specific file, no output will occur:

	```shell
	ember-template-lint app/templates/foo.hbs
	```

- when you have deleted a linted file (foo.hbs), todo files associated with that file become orphaned. You need to run `--update-todo` to cleanup todo files for that file:

	```shell
	ember-template-lint . --update-todo
	```

- when old pending system (`config.pending`) and new todo system (`.lint-todo`) co-exist, error
- when `--update-todo` ran and old pending system exists (`config.pending`), migrate old to new

### PR DESCRIPTION
**Pull request by scalvert**
Resolves #1557.

### CODE REVIEW

File: docs/todos.md
Comment (rwjblue) [inline docs/todos.md:5]:
```suggestion
It's common to leverage linting not just for syntax adherence, but also to direct developers to employ standardized patterns. As such, it's a fairly routine activity to introduce new lint rules into a codebase. This introduction, while necessary, can cause unintended friction, such as:
```

File: docs/todos.md
Comment (rwjblue) [inline docs/todos.md:9]:
```suggestion
- an "all or nothing" approach, where new rules require fixing before rollout
```

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: docs/todos.md
Comment (scalvert) [reply docs/todos.md:1]: For sure, or both?

Comment (scalvert) [reply docs/todos.md:1]: I actually just added it to the README, since configuration didn't make sense (as it's not part of configuration).

Comment (rwjblue) [reply docs/todos.md:1]: I was thinking that how to configure the default command line argument values should be documented in `docs/configuration.md`

Comment (scalvert) [reply docs/todos.md:1]: Ya we should add a bit on configuration for TODOs in `docs/todos.md` that talks about how to configure this in the package.json, env vars, options, yada yada.

**Review by rwjblue — CHANGES_REQUESTED** [review]
I think the only pending items here are to:

* Add a section to `docs/todos.md` about configuration (e.g. how you would configure `package.json`)
* Update `docs/configuration.md` to link to that new section in `docs/todos.md`

**Comment by scalvert** [comment]
Yep. I was planning on waiting to do that after the other TODO PR was merged, but I think it makes sense to add it now since that one's pretty close to being ready to go.

**Comment by rwjblue** [comment]
Sounds good, just ping me when this is updated and ready for re-review. I _think_ this is the last issue before closing https://github.com/ember-template-lint/ember-template-lint/issues/1557 ...

**Comment by scalvert** [comment]
@rwjblue ready to go.

File: docs/todos.md
Comment (rwjblue) [inline docs/todos.md]: I think _in general_ we don't expect ember-template-lint users to leverage these environment variables, think it would be ok to flip 2 and 3 (so we mention the command line flags first)? You already explain the precedence just below, so it _shouldn't_ make it more confusing to folks...

What do you think?

**Review by rwjblue — APPROVED** [review]
What do you think about also mentioning that we think folks _should_ set a default decay delay (to at least warnings)? Where might that fit? (also this seems totally fine to iterate separately)

File: docs/todos.md
Comment (scalvert) [reply docs/todos.md]: I agree. I think you and I should decide how we want to publicize this functionality. Let's do that in a follow up.

**Pull request merged**
